src/app/src/components/Submissions/Detail/Map.js

```diff
@@ -19,40 +30,118 @@ export default function Map({ submission }) {
             >
                 <MapPanes>
                     <DefaultBasemap />
-                    <MapButtons submission={submission} />
+                    <MapButtons boundary={boundary} startReview={startReview} />
+                    <Polygon shape={boundary.submission.shape} />
                 </MapPanes>
             </MapContainer>
-            <SubmissionStatusBar submission={submission} />
+            <SubmissionStatusBar status={boundary.status} />
         </>
     );
 }
 
-function MapButtons({ submission }) {
+function MapButtons({ boundary, startReview }) {
+    const navigate = useNavigate();
+    const userRole = useSelector(state => state.auth.user.role);
+
+    const canEditDraft =
+        userRole === ROLES.CONTRIBUTOR || userRole === ROLES.ADMINISTRATOR;
+    const canReview =
+        userRole === ROLES.VALIDATOR || userRole === ROLES.ADMINISTRATOR;
+
+    const getDrawButtonText = () => {
+        switch (boundary.status) {
+            case BOUNDARY_STATUS.DRAFT:
+                return canEditDraft ? 'Edit boundary' : null;
+            case BOUNDARY_STATUS.SUBMITTED:
+                return canReview ? 'Start review' : null;
+            case BOUNDARY_STATUS.IN_REVIEW:
+                return canReview ? 'Continue review' : null;
+            default:
+                return null;
+        }
+    };
+
+    const goToDrawPage = () => navigate(`/draw/${boundary.id}`);
+
+    const getDrawButtonOnClick = () => {
+        switch (boundary.status) {
+            case BOUNDARY_STATUS.DRAFT:
+                return canEditDraft ? goToDrawPage : null;
+            case BOUNDARY_STATUS.SUBMITTED:
+            case BOUNDARY_STATUS.IN_REVIEW:
+                return canReview
+                    ? () => startReview(boundary.id).unwrap().then(goToDrawPage)
+                    : null;
```

Comment:
Not sure why my earlier comment did not take, but since we don't allow IN REVIEW to create a new Review, we should adjust this such:

```suggestion
            case BOUNDARY_STATUS.DRAFT:
            case BOUNDARY_STATUS.IN_REVIEW:
                return canEditDraft || canReview ? goToDrawPage : null;
            case BOUNDARY_STATUS.SUBMITTED:
                return canReview
                    ? () => startReview(boundary.id).unwrap().then(goToDrawPage)
                    : null;
```

Reply:
How about:
```suggestion
            case BOUNDARY_STATUS.DRAFT:
                return canEditDraft ? goToDrawPage : null;
            case BOUNDARY_STATUS.IN_REVIEW:
                return canReview ? goToDrawPage : null;
            case BOUNDARY_STATUS.SUBMITTED:
                return canReview
                    ? () => startReview(boundary.id).unwrap().then(goToDrawPage)
                    : null;
```

Reply:
Yours is cleaner, I like it! Let's do that

Reply:
5fd890771f06fc0a2e57c1facf797927e8059f77